config/master_template.j2

```diff
@@ -0,0 +1,15 @@
+{%- import "generator_macros.j2" as macros with context -%}
+{{ macros.header(imports=imports, ignore=ignore) }}
+
+class {{ exercise | camel_case }}Test(unittest.TestCase):
+    {% for case in cases -%}
+    {{ test_case(case) }}
+    {% endfor %}
+
+    {% if additional_cases | length -%}
+    # Additional tests for this track
+    {% for case in additional_cases -%}
+    {{ test_case(case) }}
```

Comment:
It might be a good idea to define a macro above for `additional_test_case` that uses `test_case`, but may be overridden by an extending template if necessary. It is common for the additional cases to follow a different structure than the canonical ones.

Reply:
The code of master template does not handle nested cases. 
I added a commit to handle them correctly. Is this OK or should I still create the macro you asked for ?

Reply:
> It is common for the additional cases to follow a different structure than the canonical ones.

What I meant was that the additional test cases sometimes have a different formula. Contrived Example:
```
class WeirdAdditionalTest(unittest.TestCase):
    def test_normal_case1(self):
        self.assertEqual(property(CONSTANT1), expected)

    def test_normal_case2(self):
        self.assertEqual(property(CONSTANT2), expected)

	# Additional tests for this track
	def test_additional_case(self):
        self.assertNotEqual(CONSTANT1, CONSTANT2)
```

If you want to properly handle nested cases, I suggest something like the following, since nested `cases` arrays can be multiple levels deep:
```
{%- macro test_cases_recursive(cases) -%}
{% for case in cases -%}
	{% if "cases" in case -%}
# case["description"]
{{ test_cases_recursive(case["cases"]) }}
	{% else -%}
{{ test_case(case) }}
	{% endif -%}
{% endfor -%}
{% endmacro %}

class {{ exercise | camel_case }}Test(unittest.TestCase):
    {{ test_cases_recursive(cases) }}
```